source/server/admin/admin_factory_context.h

```diff
@@ -8,6 +8,24 @@
 namespace Envoy {
 namespace Server {
 
+class EmptyListenerInfo : public Network::ListenerInfo {
```

Comment:
why not just create a listener info impl with minimal contents?

Reply:
Didn't want to bring a dependency on extensions...

Reply:
admin already depends on listener no?  and listenre is moving back to core code soon.  we could alternately factor it out into a minimal library?  I'm just not a fan of one-off admin classes but I can look the other way if it's a pain =P